### ISSUE TITLE
Install newer CMake

### ISSUE THREAD

**Issue by madig** [pypa/manylinux#178]
Could you please supply a more modern CMake?

I'm trying to modernize the CMake build file for FreeType (https://github.com/madig/freetype2/tree/modernize-cmake-build) and want to compile it for freetype-py (https://github.com/madig/freetype-py/tree/setup-ci). 2.8.11.2 is making this harder than it should be...

**Comment by madig** [comment]
Yes, but not having to install things makes things go faster.

**Comment by gmagno** [comment]
@madig you could just create your own docker images based on the ones provided by manylinux:

    # Dockerfile x86_64
    FROM quay.io/pypa/manylinux1_x86_64
    RUN for PYBIN in /opt/python/*/bin; do "${PYBIN}/pip" install cmake; done

And for i686:

    # Dockerfile i686
    FROM quay.io/pypa/manylinux1_i686
    RUN for PYBIN in /opt/python/*/bin; do "${PYBIN}/pip" install cmake; done

**Comment by gmagno** [comment]
What about those that don't need cmake at all?

**Comment by madig** [comment]
That's what I meant by "out of scope for the project". I can install it if I need it.

**Comment by gmagno** [comment]
> That's what I meant by "out of scope for the project".

I disagree with that.

> I can install it if I need it.

I totally agree with that. 

(anway... over and out)

**Comment by bstellato** [comment]
I am having a hard time getting cmake to work by installing it with `pip install cmake`. The installation seems to go through but I am getting the error:

```
    File "/opt/python/cp36-cp36m/bin/cmake", line 6, in <module>
      from cmake import cmake
  ModuleNotFoundError: No module named 'cmake'
```

when I call `cmake --version`.

This happens to the [osqp-wheels](https://github.com/oxfordcontrol/osqp-wheels) package.  The error I am getting appears on travis at: https://travis-ci.org/oxfordcontrol/osqp-wheels/jobs/581305073#L655

The code where I install cmake is in the `pre_build` function in [this file](https://github.com/oxfordcontrol/osqp-wheels/blob/master/config.sh). It was working a few months ago. I am including @matthew-brett  since I am trying to use multibuild to do so.

**Comment by matthew-brett** [comment]
Strange - has the version of cmake changed?

I see the latest version of the cmake package only has manylinux2010 wheels, which won't work for the manylinux1 image - but your log says you're installing the previous version.

How about installing cmake with `multibuild` and `library_builders.sh` rather than via pip?

**Comment by bstellato** [comment]
> Strange - has the version of cmake changed?
>
> I see the latest version of the cmake package only has manylinux2010 wheels, which won't work for the manylinux1 image - but your log says you're installing the previous version.

Yes, it was 3.13.1 [before](https://travis-ci.org/oxfordcontrol/osqp-wheels/builds/466577585). Now it installs 3.13.3 [on travis](https://travis-ci.org/oxfordcontrol/osqp-wheels/builds/581305065) but on [pypi 3.14.4](https://pypi.org/project/cmake/#history).


> How about installing cmake with `multibuild` and `library_builders.sh` rather than via pip?

I see that there is the `get_cmake` function [here](https://github.com/matthew-brett/multibuild/blob/ed1cfaf365ed51bd309a80e60736915d3ec70d31/library_builders.sh#L152) but it installs an older version via yum (version 2.8) which we cannot use. I know it is possible to download and compile our own version but it significantly slows down CI. I could stick with the older cmake version to make it work but it would be good to have a better solution also for other users.

**Comment by lkollar** [comment]
CMake 3.14.0 and above [doesn't build on Centos 5](https://gitlab.kitware.com/cmake/cmake/issues/19086) so the last usable version on `manylinux1` is 3.13.3.

You might want to try `pip install cmake==3.13.3`.

**Comment by lkollar** [comment]
> CMake 3.14.0 and above [doesn't build on Centos 5](https://gitlab.kitware.com/cmake/cmake/issues/19086) so the last usable version on `manylinux1` is 3.13.3.

Actually, technically it could still be built using an older `libuv` present on the system, but it seems that they don't do that and just stopped providing `manylinux1` wheels.

**Comment by matthew-brett** [comment]
Can you use cmake 2.8?  As in:

```
ln -sf $(which cmake28) /usr/bin/cmake
```

**Comment by bstellato** [comment]
I have tried an older version but it [gives the same error](https://travis-ci.org/oxfordcontrol/osqp-wheels/jobs/581496075).  Unfortunately we need a newer version of cmake and 2.8 is too old. 


The weird thing is that if we run `pip install cmake` on a different python version and link it symbolically it works. But if you install it for the same python version as the one you are running, it breaks. See the hack we did for building wheels in the [osqp-python repo](https://github.com/oxfordcontrol/osqp-python/blob/master/ci/travis/build-wheels.sh). Do you know any reason why this is happening?  In our package cmake gets called from [setup.py](https://github.com/oxfordcontrol/osqp-python/blob/8e009e7dd41c44d1d2c31f92a5a0a035de6fb9f0/setup.py#L169).

**Comment by bstellato** [comment]
I fixed it by defining the function below (working for multibuild). It deletes the python script created by cmake and directly links the cmake binary to the bin folder. This happened because the shebang line of the python script included from cmake does not work well with some python versions. This is absolutely not intuitive nor easy to fix. I would strongly suggest a better way to handle cmake in the docker image.

```
function fix_cmake {
	if [ -n "$IS_OSX" ]; then
		brew update
		brew upgrade cmake || brew install cmake
	else
		# Fix cmake installation linking the appropriate binary
		pip install cmake
		rm `python -c 'import sys; print(sys.executable[:-6])'`cmake
		CMAKE_BIN=`python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`/cmake/data/bin/cmake
		ln -sf ${CMAKE_BIN} /usr/local/bin/cmake
	fi
}
```

**Comment by wenxingxing** [comment]
This helps
```dockerfile
FROM quay.io/pypa/manylinux1_x86_64:latest
RUN /opt/python/cp37-cp37m/bin/pip install -i https://pypi.tuna.tsinghua.edu.cn/simple cmake
RUN ln -s /opt/python/cp37-cp37m/bin/cmake /usr/bin/cmake

```
